test/unit/utils.py

```diff
@@ -55,3 +56,22 @@ def assertRaisesRegexp(expected_exception, expected_regexp,
         else:
             excName = str(expected_exception)
         raise unittest.TestCase.failureException("%s not raised" % excName)
+
+
+class WSGIServer(object):
+    """Start a WSGI Web server."""
+    def __init__(self, application):
+        self.application = application
+        self._thread = None
+
+        self.server = eventlet.listen(('127.0.0.1', 0))
+        (self.ip, self.port) = self.server.getsockname()
+
+    def __enter__(self):
+        self._thread = eventlet.spawn(eventlet.wsgi.server, self.server,
+                                      self.application)
+        return self
+
+    def __exit__(self, exc_ty, exc_val, tb):
+        self._thread.kill()
```

Comment:
Use try/finally to ensure the socket is closed even when killing the thread would fail for some reason.